app/pendrive_detection/pendrive_detector.py

```diff
@@ -0,0 +1,125 @@
+        return sys.platform
+
+    def find_all_pen_drives(self) -> Optional[list]:
+        """
+        Detects and lists all connected pen drives across different operating systems (Linux, Windows, macOS).
+        It determines the current operating system and delegates the task to an OS-specific function.
+        :return:
+            list - A list of detected pen drive mount points (partitions or whole drives).
+            None - if no pen drives are detected.
+        """
+        if self.check_os() == "linux":
+            return self.find_all_pen_drives_linux()
+        elif self.check_os() == "win32":
+            return self.find_all_pen_drives_win()
+        elif self.check_os() == "darwin":
+            pass  # TODO implement macos function
+
+    @staticmethod
+    def find_all_pen_drives_linux() -> Optional[list]:
+        """
+        Scans all connected and mounted drives to detect all pen drives on Linux systems.
+
+        :return:
+            list - A list of detected pen drive mount points (partitions or whole drives).
+            None - if no pen drives are detected.
+        """
+        context = pyudev.Context()
+        pen_drives = []
+
+        partition_mounts = {p.device: p.mountpoint for p in psutil.disk_partitions()}
+
+        for device in context.list_devices(subsystem="block", DEVTYPE="disk"):
+            if device.attributes.asstring("removable") == "1":
+                mount_point = partition_mounts.get(device.device_node)
+                if mount_point:
+                    pen_drives.append(mount_point)
+                else:
+                    for part in context.list_devices(
+                        subsystem="block", DEVTYPE="partition", parent=device
+                    ):
+                        mount_point = partition_mounts.get(part.device_node)
+                        if mount_point:
+                            pen_drives.append(mount_point)
+
+        return pen_drives if pen_drives else None
+
+    def find_all_pen_drives_win(self) -> Optional[list]:
+        """
+        Scans all connected and mounted drives to detect all pen drives on Windows systems.
+
+        :return:
+            list - A list of detected pen drive mount points (partitions or whole drives).
+            None - if no pen drives are detected.
+        """
+        if self.wmi_client is None:
+            self.wmi_client = wmi.WMI()
+
+        pen_drives = []
+
+        for drive in self.wmi_client.Win32_DiskDrive():
+            if drive.InterfaceType == "USB":
+                for partition in drive.associators("Win32_DiskDriveToDiskPartition"):
+                    for logical_disk in partition.associators(
+                        "Win32_LogicalDiskToPartition"
+                    ):
+                        pen_drives.append(logical_disk.DeviceID)
+
+        return pen_drives if pen_drives else None
+
+    @staticmethod
+    def find_pen_drive_with_private_key(pen_drives: list) -> Optional[str]:
```

Comment:
list is not really a good type hint you should add a type that will be inside the list for example List[str]